src/rust/inetstack/protocols/tcp/established/mod.rs

```diff
@@ -45,15 +52,48 @@ pub struct EstablishedSocket<const N: usize> {
 
 impl<const N: usize> EstablishedSocket<N> {
     pub fn new(
-        cb: SharedControlBlock<N>,
-        qd: QDesc,
-        dead_socket_tx: mpsc::UnboundedSender<QDesc>,
+        local: SocketAddrV4,
+        remote: SocketAddrV4,
         mut runtime: SharedDemiRuntime,
+        transport: SharedBox<dyn NetworkRuntime<N>>,
+        local_link_addr: MacAddress,
+        tcp_config: TcpConfig,
+        arp: SharedArpPeer<N>,
+        receiver_seq_no: SeqNumber,
+        ack_delay_timeout: Duration,
+        receiver_window_size: u32,
+        receiver_window_scale: u32,
+        sender_seq_no: SeqNumber,
+        sender_window_size: u32,
+        sender_window_scale: u8,
+        sender_mss: usize,
+        cc_constructor: CongestionControlConstructor,
+        congestion_control_options: Option<congestion_control::Options>,
+        dead_socket_tx: mpsc::UnboundedSender<QDesc>,
```

Comment:
This is a large number of arguments. We should possible give it a thought about grouping/encapsulating these in logical data structures. Note: This is not an urgent requirement, but something to think about in the long run based on how the EstablishedSocket data structure evolves/is used in future.

Reply:
Yes, definitely. I think we could certainly do a sort of clone or something of an existing socket and then set up the socket with everything that is different but I'm not sure that is a very common Rust thing to do. I think we probably want to have a socket state data structure that keeps pointers to all of these options and things, which is separate from the EstablishedSocket object but that is also not very object-oriented.